### ISSUE TITLE
Disagreement between `indentation_linter()` and `{styler}` when expression is in parentheses

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1952]
I am not sure what the tidyverse style guide dictates here, but linter conflicts with `{styler}` here for sure (cc @lorenzwalthert).

``` r
"(df <-
  dplyr::select(mtcars, wt) %>%
  dplyr::mutate(wt_sq = wt^2))" -> code

styler::style_text(code)
#> (df <-
#>   dplyr::select(mtcars, wt) %>%
#>   dplyr::mutate(wt_sq = wt^2))

lintr::lint(text = code, linters = lintr::indentation_linter())
#> <text>:2:2: style: [indentation_linter] Indentation should be 3 spaces but is 2 spaces.
#>   dplyr::select(mtcars, wt) %>%
#>  ^~
```

<sup>Created on 2023-04-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by AshesITR** [comment]
This one I actually think ours makes more sense.
What we have here is essentially a hanging indent, just no function call preceding the parenthesis that triggers it.

The following code snippets _should_ have different indentation IMO

```r
(df <-
   dplyr::select(mtcars, wt) %>%
   dplyr::mutate(wt_sq = wt^2))

df <-
  dplyr::select(mtcars, wt) %>%
  dplyr::mutate(wt_sq = wt^2)

funny(df <-
        dplyr::select(mtcars, wt) %>%
        dplyr::mutate(wt_sq = wt^2))
```

**Comment by lorenzwalthert** [comment]
In its current implementation, {styler} can by design only indent one level per *line*, no matter how many trigger tokens you have. This was a deliberate choice based on the interactive experience when working with RStudio that can't be reverted easily.
